app/script/nodes/Image/Effects/NodeHueShift.ts

```diff
@@ -0,0 +1,79 @@
+///<reference path="../NodeImageTransformation.ts" />
+
+/**
+ * @classdesc "Hue" shift the colours on an image by a given amount
+ * @author Simon, Orlando
+ */
+
+@register
+class NodeHueShift extends NodeImageTransformation {
+	private matrix: number[][];
+
+	constructor() {
+		super("Hue Shift", "Shift the colours of an image", "/Effects");
```

Comment:
Can we call this node / class "Hue Rotate" actually? I think that's typically what it's called in e.g. Photoshop